src/Uno.UWP/Storage/StorageFile.cs

```diff
@@ -33,126 +33,100 @@ private static Task<StorageFile> GetFileFromApplicationUri(CancellationToken ct,
 			=> throw new NotImplementedException();
 #endif
 
-		private readonly ImplementationBase _impl;
-
-		private StorageFile(ImplementationBase impl)
+		private StorageFile(ImplementationBase implementation)
 		{
-			_impl = impl;
-			_impl.InitOwner(this);
+			Implementation = implementation;
+			Implementation.InitOwner(this);
 		}
 
-		public string Path => _impl.Path;
+		internal ImplementationBase Implementation { get; }
+
+		public StorageProvider Provider => Implementation.Provider;
+
+		public string Path => Implementation.Path;
 
-		public string FileType => _impl.FileType;
+		public string FileType => Implementation.FileType;
 
-		public string Name => _impl.Name;
+		public string Name => Implementation.Name;
 
-		public string DisplayName => _impl.DisplayName;
+		public string DisplayName => Implementation.DisplayName;
 
-		public string ContentType => _impl.ContentType;
+		public string ContentType => Implementation.ContentType;
 
-		public DateTimeOffset DateCreated => _impl.DateCreated;
+		public DateTimeOffset DateCreated => Implementation.DateCreated;
 
 		public bool IsOfType(StorageItemTypes type)
 			=> type == StorageItemTypes.File;
 
 		public bool IsEqual(IStorageItem item)
-			=> _impl.IsEqual(item);
+			=> Implementation.IsEqual(item);
 
 		#region internal API (Task)
-		internal Task<StorageFolder> GetParent(CancellationToken ct)
-			=> _impl.GetParent(ct);
-
-		internal Task<BasicProperties> GetBasicProperties(CancellationToken ct)
-			=> _impl.GetBasicProperties(ct);
-
-		internal Task<IRandomAccessStreamWithContentType> Open(CancellationToken ct, FileAccessMode accessMode, StorageOpenOptions options)
-			=> _impl.Open(ct, accessMode, options);
 
 		internal Task<Stream> OpenStream(CancellationToken ct, FileAccessMode accessMode, StorageOpenOptions options)
-			=> _impl.OpenStream(ct, accessMode, options);

```

Comment:
No please keep those task based methods easily accessible for internal usage. They avoid lot of indirection.